File: aladdinworks44_ui/src/services/RackAssignmentService.js
import http from "../http-common"; 

class RackAssignmentService {
  getAllRackAssignments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/rackAssignment/rackAssignments`, searchDTO);
  }

  get(rackAssignmentId) {
    return this.getRequest(`/rackAssignment/${rackAssignmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/rackAssignment?field=${matchData}`, null);
  }

  addRackAssignment(data) {
    return http.post("/rackAssignment/addRackAssignment", data);
  }

  update(data) {
  	return http.post("/rackAssignment/updateRackAssignment", data);
  }
  
  uploadImage(data,rackAssignmentId) {
  	return http.postForm("/rackAssignment/uploadImage/"+rackAssignmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RackAssignmentService();

File: src/main/java/com/aladdinworks44/service/RackAssignmentService.java
package com.aladdinworks44.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks44.domain.RackAssignment;
import com.aladdinworks44.dto.RackAssignmentDTO;
import com.aladdinworks44.dto.RackAssignmentSearchDTO;
import com.aladdinworks44.dto.RackAssignmentPageDTO;
import com.aladdinworks44.dto.RackAssignmentConvertCriteriaDTO;
import com.aladdinworks44.service.GenericService;
import com.aladdinworks44.dto.common.RequestDTO;
import com.aladdinworks44.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RackAssignmentService extends GenericService<RackAssignment, Integer> {

	List<RackAssignment> findAll();

	ResultDTO addRackAssignment(RackAssignmentDTO rackAssignmentDTO, RequestDTO requestDTO);

	ResultDTO updateRackAssignment(RackAssignmentDTO rackAssignmentDTO, RequestDTO requestDTO);

    Page<RackAssignment> getAllRackAssignments(Pageable pageable);

    Page<RackAssignment> getAllRackAssignments(Specification<RackAssignment> spec, Pageable pageable);

	ResponseEntity<RackAssignmentPageDTO> getRackAssignments(RackAssignmentSearchDTO rackAssignmentSearchDTO);
	
	List<RackAssignmentDTO> convertRackAssignmentsToRackAssignmentDTOs(List<RackAssignment> rackAssignments, RackAssignmentConvertCriteriaDTO convertCriteria);

	RackAssignmentDTO getRackAssignmentDTOById(Integer rackAssignmentId);







}





